### DISCUSSION TITLE
v0.12.3b11

### DISCUSSION THREAD

**Discussion by ErikBjare** [ActivityWatch/activitywatch#972] [Releases]
These are the release notes for ActivityWatch version v0.12.3b11.

**New to ActivityWatch?** Check out the [website](https://activitywatch.net) and the [README](https://github.com/ActivityWatch/activitywatch/blob/master/README.md).

# Installation

See the [getting started guide in the documentation](https://docs.activitywatch.net/en/latest/getting-started.html).

# Downloads

 - [**Windows**](https://github.com/ActivityWatch/activitywatch/releases/download/v0.12.3b11/activitywatch-v0.12.3b11-windows-x86_64-setup.exe) (.exe, installer)
 - [**macOS**](https://github.com/ActivityWatch/activitywatch/releases/download/v0.12.3b11/activitywatch-v0.12.3b11-macos-x86_64.dmg) (.dmg)
 - [**Linux**](https://github.com/ActivityWatch/activitywatch/releases/download/v0.12.3b11/activitywatch-v0.12.3b11-linux-x86_64.zip) (.zip)

# Contributors

Thanks to everyone who contributed to this release:

@dependabot[bot], @erikbjare, @Julianoe

# Changelog

Changes since v0.12.3b10

### 🔨 Misc (1)
<p>

 - test: make integration tests also run for aw-server-rust ([`42187d8`](https://github.com/ActivityWatch/activitywatch/commit/42187d8))

</p>

### ✨ Features (1)
 - feat: added settings endpoints ([#147](https://github.com/ActivityWatch/aw-server/issues/147)) ([`520c77c`](https://github.com/ActivityWatch/aw-server/commit/520c77c))

### 🔨 Misc (1)
<p>

 - test: improved test_settings ([`60269e7`](https://github.com/ActivityWatch/aw-server/commit/60269e7))

</p>

### ✨ Features (1)
 - feat: embed webui assets into the binary ([#385](https://github.com/ActivityWatch/aw-server-rust/issues/385)) ([`26acc28`](https://github.com/ActivityWatch/aw-server-rust/commit/26acc28))

### 🐛 Fixes (5)
<p>

 - fix: return null for missing settings without 404 ([`d39f258`](https://github.com/ActivityWatch/aw-server-rust/commit/d39f258))
 - fix: rewrote settings logic to mimic aw-server-python behavior ([#428](https://github.com/ActivityWatch/aw-server-rust/issues/428)) ([`728c00e`](https://github.com/ActivityWatch/aw-server-rust/commit/728c00e))
 - fix: improve AW_WEBUI_DIR handling ([#424](https://github.com/ActivityWatch/aw-server-rust/issues/424)) ([`448312d`](https://github.com/ActivityWatch/aw-server-rust/commit/448312d))
 - fix: embed files even in debug mode, minor android build script improvements, update aw-webui ([`2e3298e`](https://github.com/ActivityWatch/aw-server-rust/commit/2e3298e))
 - fix(android): removed use of asset_path in favor of asset_resolver ([`f284213`](https://github.com/ActivityWatch/aw-server-rust/commit/f284213))

</p>

### ✨ Features (1)
 - feat: server-side settings ([#493](https://github.com/ActivityWatch/aw-webui/issues/493)) ([`1131f9b`](https://github.com/ActivityWatch/aw-webui/commit/1131f9b))

### 🐛 Fixes (4)
<p>

 - fix: removed unnecessary margin on small screens ([`573b728`](https://github.com/ActivityWatch/aw-webui/commit/573b728))
 - fix: make uncategorized notification less aggressive ([`2bcd615`](https://github.com/ActivityWatch/aw-webui/commit/2bcd615))
 - fix: support Arc browser on macOS ([#473](https://github.com/ActivityWatch/aw-webui/issues/473)) ([`cf74080`](https://github.com/ActivityWatch/aw-webui/commit/cf74080))
 - fix: wording s/ignore case/case insensitive/ ([#478](https://github.com/ActivityWatch/aw-webui/issues/478)) ([`f76721a`](https://github.com/ActivityWatch/aw-webui/commit/f76721a))

</p>

### 🐛 Fixes (2)
<p>

 - fix: fixed log path (again) ([`6adeed9`](https://github.com/ActivityWatch/aw-core/commit/6adeed9))
 - fix: change back to old log dir after breaking change in platformdirs, pinned platformdirs ([`6b3d670`](https://github.com/ActivityWatch/aw-core/commit/6b3d670))

</p>

### ✨ Features (1)
 - feat: added methods get_setting and set_setting ([`858672d`](https://github.com/ActivityWatch/aw-client/commit/858672d))

<hr /><em>This discussion was created from the release <a href='https://github.com/ActivityWatch/activitywatch/releases/tag/v0.12.3b11'>v0.12.3b11</a>.</em>

**Comment by iloveitaly** [comment]
The Rust server isn't starting for me.  Python server starts fine. Previous versions work just fine as well. Anyone else running into trouble?

**Reply by ErikBjare** [reply]
Logs?

**Reply by iloveitaly** [reply]
```
[2023-10-24 17:05:12][INFO][aw_server]: Using aw-webui assets at path None
[2023-10-24 17:05:12][INFO][aw_server::endpoints]: Starting aw-server-rust at 127.0.0.1:5600
[2023-10-24 17:05:12][WARN][rocket::launch]: 🚀 Rocket has launched from http://127.0.0.1:5600
[2023-10-24 17:05:15][WARN][rocket::response::responder::_]: Response was `None`.
[2023-10-24 17:05:15][WARN][rocket::server::_]: No 404 catcher registered. Using Rocket default.
2023-10-24 17:05:15 [aw-watcher-window-macos] [ERROR] tab title diff: ‎127.0.0.1:5600, window title: Untitled
2023-10-24 17:05:15 [aw-watcher-window-macos] [ERROR] tab title diff: 404 Not Found, window title: ‎127.0.0.1:5600
[2023-10-24 17:05:15][WARN][rocket::response::responder::_]: Response was `None`.
[2023-10-24 17:05:15][WARN][rocket::server::_]: No 404 catcher registered. Using Rocket default.
[2023-10-24 17:05:18][WARN][aw_server::endpoints::bucket::_]: Data guard `Json < Event >` failed: Parse("{\"timestamp\":\"\",\"duration\":0,\"data\":{\"title\":\"49\",\"session_name\":\"49\",\"window_name\":\"aw-qt\",\"pane_title\":\"/Applications/ActivityWatch.app/Contents/MacOS/aw-qt\",\"pane_current_command\":\"aw-qt\",\"pane_current_path\":\"/Users/mike\"}}", Error("premature end of input", line: 1, column: 15)).
[2023-10-24 17:05:18][WARN][rocket::server::_]: No 422 catcher registered. Using Rocket default.
[2023-10-24 17:05:19][WARN][rocket::response::responder::_]: Response was `None`.
[2023-10-24 17:05:19][WARN][rocket::server::_]: No 404 catcher registered. Using Rocket default.
2023-10-24 17:05:19 [aw-watcher-window-macos] [ERROR] tab title diff: ‎127.0.0.1:5600, window title: 404 Not Found
```

And here's the HTTP response

```
❯ http -vvv http://127.0.0.1:5600
GET / HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: 127.0.0.1:5600
User-Agent: HTTPie/3.2.2



HTTP/1.1 404 Not Found
content-length: 383
content-type: text/html; charset=utf-8
date: Tue, 24 Oct 2023 23:06:18 GMT
permissions-policy: interest-cohort=()
server: Rocket
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>404 Not Found</title>
</head>
<body align="center">
    <div role="main" align="center">
        <h1>404: Not Found</h1>
        <p>The requested resource could not be found.</p>
        <hr />
    </div>
    <div role="contentinfo" align="center">
        <small>Rocket</small>
    </div>
</body>
</html>


Elapsed time: 0.002444542s
```

**Reply by ErikBjare** [reply]
@iloveitaly Was also reported here: https://github.com/ActivityWatch/activitywatch/issues/974

Just a simple build issue, rolling v0.12.3b12 now!

**Reply by ErikBjare** [reply]
Give the new release a try, and let me know if there are any issues: https://github.com/ActivityWatch/activitywatch/releases/tag/v0.12.3b12

**Reply by iloveitaly** [reply]
Works great! Thanks.
